### PR TITLE
PA-2527 add new erp note type to approval form and add that note to ERP notifications.

### PR DESCRIPTION
**Pull request by devinleighsmith**

### CODE REVIEW

File: backend/dal/Migrations/v01.06.01/Down/PreDown/01-NotificationTemplates.sql [hunk 1]
```diff
@@ -0,0 +1,728 @@
+UPDATE dbo.[NotificationTemplates]
+SET [Body] = '<html><head><title>@Model.Environment.Title</title></head><body><p>Dear @Model.ToAgency.AddressTo,</p><p>Your project @Model.Project.ProjectNumber has been approved. Signin to <a href="@Model.Environment.Uri">PIMS</a> to review the progress.</p><p>Sincerely Real Property Division</p></body></html>'
+WHERE [Id] = 4
+
+UPDATE dbo.[NotificationTemplates]
+SET [Body] = '
+@using System.Linq
```
Comment (agalbrai) [inline backend/dal/Migrations/v01.06.01/Down/PreDown/01-NotificationTemplates.sql:7]: so is this an html file in the database?

Comment (Fosol) [reply backend/dal/Migrations/v01.06.01/Down/PreDown/01-NotificationTemplates.sql:7]: Technically it's a Razor file in the database.

File: backend/dal/Migrations/v01.06.01/Up/PreUp/01-NotificationTemplates.sql [hunk 2]
```diff
@@ -0,0 +1,742 @@
...
+<p><b>Should there be no interest in the property detailed below from your Ministry or any SUCH Sector Organization or BPS Entity that your Ministry is responsible for, please respond to confirm. </b></p>
+<p>This email is to advise that the following property has been identified as surplus by the current holder of the property and is available for redeployment if there is a need by your Ministry, SUCH Sector Organization or BPS Entity:</p>
+<p>
+    <ol>
+        @foreach (var property in Model.Project.Properties)
+        {
+        <li>
+            @if (property.PropertyType == PropertyTypes.Land)
+            {
+                var appraised = property.Parcel.Evaluations.OrderByDescending(e => e.Date).FirstOrDefault(e => e.Key == EvaluationKeys.Appraised);
```
Comment (Fosol) [inline backend/dal/Migrations/v01.06.01/Up/PreUp/01-NotificationTemplates.sql:28]: This value will never be populated on a property anymore.

File: backend/dal/Migrations/v01.06.01/Up/PreUp/01-NotificationTemplates.sql [hunk 3]
```diff
@@ -0,0 +1,742 @@
...
+                    }
+                    else if (assessed != null && assessed.Value > 0)
+                    {
+                        <span>Assessed Value: @assessed.Value.ToString("C", new CultureInfo("en-US")) (@assessed.Date.Year)</span>
+                    }
+                </span>
+            }
+            else
+            {
+                var appraised = property.Building.Evaluations.OrderByDescending(e => e.Date).FirstOrDefault(e => e.Key == EvaluationKeys.Appraised);
```
Comment (Fosol) [inline backend/dal/Migrations/v01.06.01/Up/PreUp/01-NotificationTemplates.sql:51]: This value will never be populated on a property anymore.

File: backend/dal/Migrations/v01.06.01/Up/PreUp/01-NotificationTemplates.sql (hunk 2 above)
Comment (Fosol) [reply backend/dal/Migrations/v01.06.01/Up/PreUp/01-NotificationTemplates.sql:28]: Same goes for all templates.

File: backend/dal/Migrations/v01.06.01/Up/PreUp/01-NotificationTemplates.sql (hunk 3 above)
Comment (Fosol) [reply backend/dal/Migrations/v01.06.01/Up/PreUp/01-NotificationTemplates.sql:51]: Same goes for all templates.

File: frontend/src/constants/noteTypes.ts [hunk 4]
```diff
@@ -46,4 +46,6 @@ export enum NoteTypes {
   SplRemoval = 20,
   /** Notes related to the provided documentation. */
   Documentation = 21,
+  /** Notes displayed on ERP notifications. */
+  ErpNotification = 22,
```
Comment (Fosol) [inline frontend/src/constants/noteTypes.ts:50]: This structure has come in handy for all the notes they keep adding.

File: frontend/src/features/projects/common/components/ProjectNotes.tsx [hunk 5]
```diff
@@ -84,3 +85,26 @@ export const PublicNotes = ({
     />
   );
 };
+
+export const ErpNotificationNotes = ({
```
Comment (Fosol) [inline frontend/src/features/projects/common/components/ProjectNotes.tsx:89]: Document

**Review by Fosol — APPROVED** [review]

File: backend/dal/Migrations/v01.06.01/Up/PreUp/01-NotificationTemplates.sql (hunk 2 above)
Comment (devinleighsmith) [reply backend/dal/Migrations/v01.06.01/Up/PreUp/01-NotificationTemplates.sql:28]: I have another story to deal with this, but I suppose I can do both in this one migration.

Comment (Fosol) [reply backend/dal/Migrations/v01.06.01/Up/PreUp/01-NotificationTemplates.sql:28]: It isn't necessary

Comment (devinleighsmith) [reply backend/dal/Migrations/v01.06.01/Up/PreUp/01-NotificationTemplates.sql]: @Fosol PA-2526(financials linked to properties not projects in erp notifications), PA-2527(new erp note type in erp notifications) and PA-1698 (do not default project financials to $0) are all tightly coupled, so I've update the review to contain fixes for all of those issues.

File: backend/dal/Migrations/v01.06.01/Down/PreDown/01-NotificationTemplates.sql (hunk 1 above)
Comment (LawrenceLau2020) [reply backend/dal/Migrations/v01.06.01/Down/PreDown/01-NotificationTemplates.sql:7]: Hi Devin, just to let you know that I just had a PR which is already using the v01.06.01 number.

**Review by LawrenceLau2020 — CHANGES_REQUESTED** [review]
Hi Devin, just to let you know that my most recent PR, PA-1742 is already using the migration version number v01.06.01.

**Review by LawrenceLau2020 — APPROVED** [review]

**Comment by LawrenceLau2020** [comment]
You might want to get a second pair of eyes to do the code review, I just removed the "block".

**Pull request merged**